src/parser.rs

```diff
@@ -435,21 +435,15 @@ impl Parser {
             } else {
                 self.expect_keyword("TRUNCATE")?;
                 let filler = match self.peek_token() {
-                    Some(tok) => {
-                        if tok == Token::make_keyword("WITH")
-                            || tok == Token::make_keyword("WITHOUT")
-                        {
-                            None
-                        } else {
-                            Some(Box::new(self.parse_expr()?))
-                        }
+                    Some(Token::Word(kw)) if kw.keyword == "WITH" || kw.keyword == "WITHOUT" => {
+                        None
                     }
-                    None => None,
+                    Some(Token::SingleQuotedString(_)) => Some(Box::new(self.parse_expr()?)),
```

Comment:
If you want to only support literal string filler, use `parse_literal_string()` and make it a `filler: Option<String>`.

Reply:
Ah thanks. That seems [to be to spec](https://jakewheat.github.io/sql-overview/sql-2016-foundation-grammar.html#listagg-truncation-filler), right?

Reply:
I doubt this is a good idea though, as I bet someone will go and ask us to at least extend support to `N'...'` literals and then Postgres will accept expressions here...

Reply:
Hm, it does seem like we should be a little more concrete here: not every expression is valid in this position.

Also for context, I don't think Postgres supports LISTAGG altogether (or maybe I'm just failing to find it).

Reply:
> not every expression is valid in this position

The same can be said about 'delimiter'. Omitting which is also not "valid", yet you yourself wrote "we choose to make the separator optional as this provides the more general implementation". The same logic can be applied here: "we choose to allow any expression not starting with `WITH` or `WITHOUT` as a `<filler>` as this provides the more general implementation".

I [know](https://github.com/andygrove/sqlparser-rs/issues/169#issuecomment-634108887) Postgres doesn't support LISTAGG at the moment. I was making a prediction that someone will deviate from the spec by supporting something other than `'string'` literals here. I brought up Postgres, as I have the impression they like to make things more general whenever possible...

I see you decided to keep it an `Expr` in the AST, but restrict the allowed tokens to certain kinds of literals (`''`, `N''`, `X''`). OK, let's do it this way.

Reply:
You're right that the same logic could be applied here, re: the more general implementation. What occurred to me however is this question: What possible values actually make sense here? In other words, what non-string filler would work in practice? I couldn't really think of any, but I very well may be overlooking the obvious. 🙂

I'm more than happy to do what you think is best, so if you decide you'd prefer another implementation then I will swap that in for what we have here.

Reply:
Your argument makes perfect sense, but in practice "LISTAGG(ALL" shouldn't exist, yet in SQL it does ;p

`'.'+'.'+'.'` is a _string_ filler just not a literal string.

But as I said, the way you did it is fine by me. If/when someone comes around wanting to lift this restriction, it would be a trivial change, rather than if you followed my original idea of changing `Expr` to `String` in the AST.